src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx

```diff
@@ -0,0 +1,577 @@
+import React, { useState, useEffect } from 'react'
+import { showError, Progressing, ErrorScreenManager, ConfirmationDialog } from '../../common'
+import docker from '../../../assets/icons/misc/docker.svg'
+import { ReactComponent as DeployButton } from '../../../assets/icons/ic-deploy.svg'
+import { InstalledAppInfo } from '../../external-apps/ExternalAppService'
+import { ServerErrors } from '../../../modals/commonTypes'
+import { Moment12HourFormat, URLS } from '../../../config'
+import CodeEditor from '../../CodeEditor/CodeEditor'
+import moment from 'moment'
+import Tippy from '@tippyjs/react'
+import '../../app/details/cIDetails/ciDetails.scss'
+import YAML from 'yaml'
+import './chartDeploymentHistory.scss'
+import MessageUI from '../common/message.ui'
+import { toast } from 'react-toastify'
+import { useHistory, useRouteMatch } from 'react-router'
+import CDEmptyState from '../../app/details/cdDetails/CDEmptyState'
+import {
+    ChartDeploymentDetail,
+    ChartDeploymentHistoryResponse,
+    ChartDeploymentManifestDetail,
+    getDeploymentHistory,
+    getDeploymentManifestDetails,
+    rollbackApplicationDeployment,
+    RollbackReleaseRequest,
+} from './chartDeploymentHistory.service'
+import IndexStore from '../appDetails/index.store'
+
+interface DeploymentManifestDetail extends ChartDeploymentManifestDetail {
+    loading?: boolean
+    error?: boolean
+    errorCode?: number
+    isApiCallInProgress?: boolean
+}
+
+function ChartDeploymentHistory({
+    appId,
+    appName,
+    isExternal,
+}: {
+    appId: string
+    appName?: string
+    isExternal: boolean
+}) {
+    const [isLoading, setIsLoading] = useState(true)
+    const [errorResponseCode, setErrorResponseCode] = useState(undefined)
```

Comment:
Why do we need to set it undefined?

Reply:
Not sure why it's been set as undefined. Fixed it.